pages/curriculum/[lesson].tsx

```diff
@@ -15,13 +17,26 @@ import _ from 'lodash'
 
 const Challenges: React.FC<QueryDataProps<AppData>> = ({ queryData }) => {
   const { lessons, session, alerts } = queryData
-  const userSubmissions: UserSubmission[] = _.get(session, 'submissions', [])
-  const lessonStatus: LessonStatus[] = _.get(session, 'lessonStatus', [])
   const router = useRouter()
   const currentlessonId = router.query.lesson as string
+  if (!session) {
+    router.push('/login')
+    return <LoadingSpinner />
+  }
+  if (!lessons || !alerts) {
+    return (
+      <Error title="Internal server error" message="Bad data" src="/500.png" />
```

Comment:
Shouldn't this mean the user is not logged in?

Reply:
No session means that user is not logged in. This case happens if there is a session but no lessons or no alerts.

Reply:
Wouldn't it better to redirect to login if this is the case?

Reply:
I think it would  be more confusing. Since there is nothing wrong with session already authorized user could be redirected to login page. He relogins, login redirects him to curriculum page, no lessons/alerts, he is on log in page again. 
This way he'll get a 500 error screen and won't waste time logging in/logging out.